addons/bus/static/src/js/longpolling_bus.js

```diff
@@ -0,0 +1,252 @@
+    ERROR_RETRY_DELAY: 10000, // 10 seconds
+    POLL_ROUTE: '/longpolling/poll',
+
+    // properties
+    _activated: false,
+    _lastNotificationID: 0,
+    _isOdooFocused: true,
+
+    /**
+     * @override
+     */
+    init: function (parent, params) {
+        this._super.apply(this, arguments);
+        this._id = _.uniqueId('bus');
+        this._options = {};
+        this._channels = [];
+
+        // bus presence
+        this._lastPresenceTime = new Date().getTime();
+        this._lastPartnersPresenceCheck = this._lastPresenceTime;
+        $(window).on("focus." + this._id, this._onFocusChange.bind(this, {focus: true}));
+        $(window).on("blur." + this._id, this._onFocusChange.bind(this, {focus: false}));
+        $(window).on("unload." + this._id, this._onFocusChange.bind(this, {focus: false}));
+
+        $(window).on("click." + this._id, this._onPresence.bind(this));
+        $(window).on("keydown." + this._id, this._onPresence.bind(this));
+        $(window).on("keyup." + this._id, this._onPresence.bind(this));
+    },
+    /**
+     * @override
+     */
+    destroy: function () {
+        var self = this;
+        this.stopPolling();
+        $(window).off("focus." + this._id);
+        $(window).off("blur." + this._id);
+        $(window).off("unload." + this._id);
+        _.each('click,keydown,keyup'.split(','), function (evtype) {
+            $(window).off(evtype + "." + self._id);
+        });
+    },
+    //--------------------------------------------------------------------------
+    // Public
+    //--------------------------------------------------------------------------
+    /**
+     * Register a new channel to listen on the longpoll (ignore if already
+     * listening on this channel).
+     * Aborts a pending longpoll, in order to re-start another longpoll, so
+     * that we can immediately get notifications on newly registered channel.
+     *
+     * @param  {string} channel
+     */
+    addChannel: function (channel) {
+        if (this._channels.indexOf(channel) === -1) {
+            this._channels.push(channel);
+            if (this._pollRpc) {
+                this._pollRpc.abort();
+            } else {
+                this.startPolling();
+            }
+        }
+    },
+    /**
+     * Unregister a channel from listening on the longpoll.
+     *
+     * Aborts a pending longpoll, in order to re-start another longpoll, so
+     * that we immediately remove ourselves from listening on notifications
+     * on this channel.
+     *
+     * @param  {string} channel
+     */
+    deleteChannel: function (channel) {
+        var index = this._channels.indexOf(channel);
+        if (index !== -1) {
+            this._channels.splice(index, 1);
+            if (this._pollRpc) {
+                this._pollRpc.abort();
+            }
+        }
+    },
+    /**
+     * Tell whether odoo is focused or not
+     *
+     * @returns  {Boolean}
```

Comment:
Remove double whitespace between `@param` and `{Boolean}` (also: `{boolean}` because primitive type value is returned)